testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js

```diff
@@ -254,19 +256,40 @@ testModule('API', moduleConfig, () => {
         assert.strictEqual(this.instance.option('value'), '<p>cbaTest 1</p><p>Test 2</p><p>Test 3</p>', 'redo operation');
     });
 
-    test('clearHistory', function(assert) {
+    test('value should not changed when "undo" called after call "clearHistory"', function(assert) {
         this.createEditor();
         this.instance.insertText(0, 'a');
         this.clock.tick(1000);
-        this.instance.insertText(0, 'b');
-        this.clock.tick(1000);
-        this.instance.insertText(0, 'c');
+
+        this.instance.clearHistory();
+        this.instance.undo();
+
+        assert.strictEqual(this.instance.option('value'), '<p>aTest 1</p><p>Test 2</p><p>Test 3</p>', 'value is actual');
+    });
+
+    test('undo button should have disabled state after call "clearHistory"', function(assert) {
+        this.options.toolbar = { items: ['undo'] };
+        this.createEditor();
+        this.instance.insertText(0, 'a');
         this.clock.tick(1000);
+        const $undoButton = $('#htmlEditor').find(`.${DX_BUTTON_CLASS}`);
 
         this.instance.clearHistory();
+
+        assert.strictEqual($undoButton.hasClass(DISABLED_STATE_CLASS), true, 'undo button disabled');
+    });
+
+    test('redo button should have disabled state after call "undo"->"clearHistory"', function(assert) {
+        this.options.toolbar = { items: ['redo'] };
+        this.createEditor();
+        this.instance.insertText(0, 'a');
+        this.clock.tick(1000);
+        const $redoButton = $('#htmlEditor').find(`.${DX_BUTTON_CLASS}`);
+
         this.instance.undo();
+        this.instance.clearHistory();
 
-        assert.strictEqual(this.instance.option('value'), '<p>cbaTest 1</p><p>Test 2</p><p>Test 3</p>', 'history is empty');
```

Comment:
hm, we kept such a test for undo, but removed for redo. Let's keep it as well?

Reply:
I did not remove step with redo but I will add test for redo call